### PR TITLE
Viewer update optimizations, and reorganization, docstrings, typing, documentation

### LINKED ISSUE #15: Update docstrings, typing, and documentation

### PR DESCRIPTION
**Pull request by klimaj**
The aim of this PR is to:

- Fix #15 
- Add unit tests for `nglview`
- For updating a pose in the viewer (in the `makeBundle` preset), currently the viewer updates after removing the old model, then updates again after adding the new model, causing the viewer to blink to background when changing poses with widgets. In this PR, `py3Dmol` now only updates after removing the old pose and adding the new, so the viewer snaps to the next pose. In `nglview`, I'm not aware of a viewer update function, so the viewer now adds the new model in and updates, then deletes the old model and updates (so there at least isn't a blink of background).
- Add docstrings
- Add typing
- Add Jupyter notebook use case examples in `README.md`
- Catch `import pyrosetta` `ImportError` exception with install instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +27,198 @@ Please open a pull request with an updated unit test from a custom branch.
...
+v.set_modules([viewer3d.setStyle(), viewer3d.setDisulfides()])
+
+for h in range(5):
+    for i in range(5):
+        for j in range(5):
+            backrub.apply(pose)
+            minimize.apply(pose)
+        v.add_pose(pose.clone(), index=h, update_viewer=False)
+v.show()
+```
```
Comment (ajasja) [inline README.md:225]: Nice. I think in the future the examples could also go into stand alone notebooks.

**Review by ajasja — APPROVED** [review]
Make bundle works nicely with py3Dmol, but with nglviewer the same problems of slow sequential updating are present.

File: README.md (hunk 1 above)
Comment (klimaj) [reply README.md:225]: That's a good idea - opened issue #19

**Comment by klimaj** [comment]
@ajasja I also still notice sometimes a blinking with `nglview`, but it looks more stable using `continuous_update=False` since the viewer otherwise can get overwhelmed with too many visualization updates (often that the user doesn't need to see when making big shifts in parameter values).

**Pull request merged**